### ISSUE TITLE
Rule `no-textarea-mustache` cannot difference `<Textarea>` component

### ISSUE THREAD

**Issue by qmhc** [vuejs/eslint-plugin-vue#2023]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** `^8.24.0`
- **eslint-plugin-vue version:** `^9.5.1`
- **Node version:** `16.17.1`
- **Operating System:** `Windows 10`

**Please show your full configuration:**

```json5
{
  extends: [
    'plugin:vue/vue3-recommended',
    // 'standard',
    // 'plugin:jsonc/recommended-with-jsonc',
    // 'plugin:yml/standard',
    // 'plugin:markdown/recommended',
    '@vue/typescript/recommended'
  ],
  env: {
    es6: true,
    browser: true,
    node: true
  },
  parser: 'vue-eslint-parser',
  parserOptions: {
    ecmaVersion: 'latest'
  },
  reportUnusedDisableDirectives: true
}
```

**What did you do?**	

```vue
<template>
  <Textarea value="Almost" :max-length="10">
    <template #count="{ value }">
      {{ `${value.length}/10` }}
    </template>
  </Textarea>
</template>
```

**What did you expect to happen?**

No error occurred.

**What actually happened?**

Get a error:

```log
error  Unexpected mustache. Use 'v-model' instead  vue/no-textarea-mustache
```

**Repository to reproduce this issue**

You can add a valid test in [here](https://github.com/vuejs/eslint-plugin-vue/blob/master/tests/lib/rules/no-textarea-mustache.js#L25):

```json5
{
  filename: 'test.vue',
  code: '<template><div><Textarea>{{text}}</Textarea></div></template>'
}
```

Then test it and it will fail.
